signprot/views.py

```diff
@@ -511,21 +502,272 @@ def fams_tab():
+            #print(prot.web_links.index.filter(web_resource__name="Guide To Pharmacology", protein__entry_name=prot.entry_name))
+            # if uniprot_links.count() > 0:
+            #     protein_data[prot.id]['uniprot_link'] = uniprot_links
+
+            # gtop_links = prot.web_links.filter(web_resource__slug='gtop')
+            # if gtop_links.count() > 0:
+            #     protein_data[prot.id]['gtp_link'] = prot.web_links.filter(web_resource__slug='gtop')
+
+            gprotein_families = ["Gs family", "Gi/Go family", "Gq/G11 family", "G12/G13 family"]
+            for gprotein in gprotein_families:
+                if prot.entry_name in signaling_data and gprotein in signaling_data[prot.entry_name]:
+                    if signaling_data[prot.entry_name][gprotein] == "primary":
+                        protein_data[prot.id][gprotein] = "1'"
+                    elif signaling_data[prot.entry_name][gprotein] == "secondary":
+                        protein_data[prot.id][gprotein] = "2'"
+                    else:
+                        protein_data[prot.id][gprotein] = "-"
+                else:
+                    protein_data[prot.id][gprotein] = "-"
+
+            protein_data[prot.id]['gs'] = protein_data[prot.id][gprotein_families[0]]
+            protein_data[prot.id]['gio'] = protein_data[prot.id][gprotein_families[1]]
+            protein_data[prot.id]['gq11'] = protein_data[prot.id][gprotein_families[2]]
+            protein_data[prot.id]['g1213'] = protein_data[prot.id][gprotein_families[3]]
+
+        couplings2 = ProteinGProteinPair.objects.filter(source__in=["Inoue", "Bouvier"])\
+            .filter(g_protein_subunit__family__slug__startswith="100_001").order_by("g_protein_subunit__family__slug", "source")\
+            .prefetch_related('g_protein_subunit__family', 'g_protein')
+
+        coupling_headers = ProteinGProteinPair.objects.filter(source__in=["Inoue", "Bouvier"])\
+            .filter(g_protein_subunit__family__slug__startswith="100_001")\
+            .order_by("g_protein_subunit__family__slug", "source").distinct("g_protein_subunit__family__slug")\
+            .values_list("g_protein_subunit__family__name", "g_protein_subunit__family__parent__name")
+
+        coupling_header_names = {}
+        coupling_reverse_header_names = {}
+        coupling_placeholder = {}
+        coupling_placeholder2 = {}
+        coupling_placeholder3 = {}
+        for name in coupling_headers:
+            if name[1] not in coupling_header_names:
+                coupling_header_names[name[1]] = []
+                coupling_placeholder3[name[1]] = []
+            coupling_reverse_header_names[name[0]] = name[1]
+            coupling_header_names[name[1]].append(name[0])
+            coupling_placeholder[name[0]] = "--"
+            coupling_placeholder2[name[0]] = []
+
+        dictotemplate = {}
+        sourcenames = set()
+        for pair in couplings2:
+            if pair.protein_id not in dictotemplate:
```

Comment:
To make it more succinct you could consider making another for loop that loops over ["logemax50", "pec50", "max"]. That way you can reduce the lines by approx ~2/3